src/provisioning.py

```diff
@@ -24,10 +28,9 @@ def prepare_data(self):
                                    [self.ipa_hyphenation_dict.inserted("  "+x+"  ").strip() for x in v]
                                for k, v in self.pronunciations.items()}
 
-    def post_to_db(self, word):
-        spelling = word.replace("-","").lower()
-        syllables = [x for x in word.split("-") if x != '']
-
+    def post_word_to_db(self, word):
+        spelling = word.replace("-","")
+        syllables = word.split("-")
         for pronunciation in self.pronunciations[word]:
```

Comment:
parasitic hyphens can occur at the beginning of words, this is what my code takes care of...